### PR TITLE
chore(storage): gRPC object/bucket empty update parity with JSON

### LINKED ISSUE #7685: storage: address grpc/http difference in response to empty bucket update
gRPC returns an error, HTTP allows it as a no-op

- Object update has the same issue

### PR DESCRIPTION
**Pull request by BrennaEpp**
Fixes #7685

### CODE REVIEW

File: storage/grpc_client.go [hunk 1]
```diff
@@ -592,6 +599,16 @@ func (c *grpcStorageClient) UpdateObject(ctx context.Context, params *updateObje
 
 	req.UpdateMask = fieldMask
 
+	if len(fieldMask.Paths) < 1 {
+		// Nothing to update. To maintain consistency with JSON, we must still
+		// update the object because metageneration and other fields are
+		// updated even on an empty update.
+		// gRPC will fail if the fieldmask is empty, so instead we add an
+		// output-only field to the update mask. Output-only fields are (and must
+		// be - see AIP 161) ignored, but allow us to send an empty update.
```
Comment (tritone) [inline storage/grpc_client.go:608]: We should maybe clarify that this is intended behavior. Feels inconsistent that empty updates are barred but this is permitted.

File: storage/integration_test.go [hunk 2]
```diff
@@ -3879,15 +3898,6 @@ func TestIntegration_CustomTime(t *testing.T) {
 		if _, err := obj.Update(ctx, ObjectAttrsToUpdate{CustomTime: earlierTime}); err == nil {
 			t.Fatalf("backdating CustomTime: expected error, got none")
 		}
-
-		// Zero value for CustomTime should be ignored. Set TemporaryHold so that
-		// we don't send an empty update request, which is invalid for gRPC.
-		if _, err := obj.Update(ctx, ObjectAttrsToUpdate{TemporaryHold: false}); err != nil {
-			t.Fatalf("empty update: %v", err)
-		}
-		if err := checkCustomTime(laterTime); err != nil {
```
Comment (tritone) [inline storage/integration_test.go:3888]: I think we should still check that custom time doesn't change after an empty update right?

Comment (BrennaEpp) [reply storage/integration_test.go]: Yeah... I figured we test empty updates elsewhere in several places, but I guess we don't actually set a CustomTime before the empty update, so I've added this back.

File: storage/grpc_client.go (hunk 1 above)
Comment (BrennaEpp) [reply storage/grpc_client.go]: Added the reasoning behind why this is permitted to the comment.

**Review by tritone — APPROVED** [review]

**Pull request merged**